internal/consoleplugin/config.go

```diff
@@ -0,0 +1,223 @@
+package consoleplugin
+
+import (
+	"fmt"
+
+	consolev1alpha1 "github.com/openshift/api/console/v1alpha1"
+	appv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Config holds configuration for the logging console plugin.
+// Construct with NewConfig.
+type Config struct {
+	Name      string
+	Namespace string
+	Image     string
+	LokiName  string
+	LokiPort  int32
+	CreatedBy string
+
+	nginxPort, defaultMode, replicas int32
+}
+
+// NewConfig returns a config with default settings.
+// Modify exported fields to change settings.
+func NewConfig(namespace, createdBy string) *Config {
+	return &Config{
+		Name:      "logging-view-plugin",
+		Namespace: namespace,
+		CreatedBy: createdBy,
+		Image:     "quay.io/gbernal/logging-view-plugin:latest", // FIXME
+		LokiName:  "loki",                                       // FIXME
+		LokiPort:  3100,
+
+		nginxPort:   9443,
+		defaultMode: 420,
+		replicas:    1,
+	}
+}
+
+func (cf *Config) secret() string              { return cf.Name + "-secret" }
+func (cf *Config) configMapName() string       { return cf.Name + "-conf" }
+func (cf *Config) selector() map[string]string { return map[string]string{"app": cf.Name} }
+
+const (
+	labelApp             = "app"
+	labelAppK8sName      = "app.kubernetes.io/name"
+	labelAppK8sCreatedBy = "app.kubernetes.io/created-by"
+)
+
+// setup adds common settings for all objects
+func (cf *Config) setup(o client.Object) {
+	l := o.GetLabels()
+	if l == nil {
+		l = map[string]string{}
+	}
+	l[labelApp] = cf.Name
+	l[labelAppK8sName] = cf.Name
+	l[labelAppK8sCreatedBy] = cf.CreatedBy
+	o.SetLabels(l)
+}
+
+func (cf *Config) configMap() *corev1.ConfigMap {
```

Comment:
Please use the utility functions in our runtime package in lieu of writing a new one

Reply:
This is not a generic utility function, it populates the configmap with data required for the console. In this case it seems simpler to just create a corev1.ConfigMap{...} with the necessary data than to use the more indirect builders.

Reply:
This is the function of which I speak https://github.com/openshift/cluster-logging-operator/blob/master/internal/k8shandler/configmap.go#L11

Reply:
Can we correct it "WHEN" and "IF" it moves and reduce the code duplication in the code base?

Reply:
See latest comments..